### PR TITLE
ceph: auto-detect rgw pool prefix for external script

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

When running the script, we will try to auto-detect the prefix of the
rgw pool and not default it anymore to "default".

Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by leseb** [comment]
@aruniiird can you please look at the unit test failure and recommend an approach? Thanks

File: cluster/examples/kubernetes/ceph/create-external-cluster-resources.py [hunk 1]
```diff
@@ -287,22 +288,47 @@ def __init__(self, arg_list=None):
...
+
+        found = False
+        for pool in json_out:
+            if '.rgw.meta' in pool['poolname']:
+                pool = pool['poolname']  # type: str
+                # extract prefix name
+                rc = re.compile(r'.+?(?=\.rgw\.meta)')
+                self._arg_parser.rgw_pool_prefix = rc.findall(pool)[0]
+                found = True
+                return self._arg_parser.rgw_pool_prefix
```
Comment (aruniiird) [inline cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:326]: Do we require `found` bool variable as we are returning immediately after finding the rgw-prefix?

File: cluster/examples/kubernetes/ceph/create-external-cluster-resources.py [hunk 2]
```diff
@@ -287,22 +288,47 @@ def __init__(self, arg_list=None):
...
+        # if there is an unsuccessful attempt,
+        if ret_val != 0 or len(json_out) == 0:
+            raise ExecutionFailureException(
+                "'osd lspools' command failed.\n" +
+                "Error: {}".format(err_msg if ret_val != 0 else self.EMPTY_OUTPUT_LIST))
+
+        found = False
+        for pool in json_out:
+            if '.rgw.meta' in pool['poolname']:
+                pool = pool['poolname']  # type: str
```
Comment (aruniiird) [inline cluster/examples/kubernetes/ceph/create-external-cluster-resources.py:321]: Is it ok to make a new variable name here (like `poolname`) ?

**Review by aruniiird — CHANGES_REQUESTED** [review]
Before the change, we used to silently ignore `rgw_pool_prefix`, if there is no `rgw_endpoint` is provided and print the output (without `ceph-rgw` storageclass in the output). But with the new change, we are hitting the `Exception`  even for the default run.
Is this expected? or Should we keep the backward compatibility?

**Comment by leseb** [comment]
`default` seems to be the default prefix so this might not be needed. Tools like Rook, Ceph-Ansible and cephadm create a default multisite config and thus we always have a prefix. Closing for now.

**Pull request closed without merging**